Framework/HistogramData/test/HistogramIteratorTest.h

```diff
@@ -0,0 +1,329 @@
+
+  void
+  test_iterate_over_histogram_frequency_variances_when_histogram_has_frequencies() {
+    Histogram hist(BinEdges{1, 2, 3, 5}, Frequencies{2, 3, 4});
+    auto expectedFrequencyVariances = hist.frequencyVariances();
+
+    TS_ASSERT(std::equal(hist.begin(), hist.end(),
+                         expectedFrequencyVariances.begin(),
+                         [](const HistogramItem &item, const double &variance) {
+                           return item.frequencyVariance() == variance;
+                         }));
+  }
+
+  void
+  test_iterate_over_histogram_frequency_std_when_histogram_has_frequencys() {
+    Histogram hist(BinEdges{1, 2, 3, 5}, Counts{2, 3, 4});
+    auto expectedFrequencyStd = hist.frequencyStandardDeviations();
+
+    TS_ASSERT(std::equal(hist.begin(), hist.end(), expectedFrequencyStd.begin(),
+                         [](const HistogramItem &item, const double &sigma) {
+                           return item.frequencyStandardDeviation() == sigma;
+                         }));
+  }
+
+  void
+  test_iterate_over_histogram_frequency_std_when_histogram_has_frequencies() {
+    Histogram hist(BinEdges{1, 2, 3, 5}, Frequencies{2, 3, 4});
+    auto expectedFrequencyStd = hist.frequencyStandardDeviations();
+
+    TS_ASSERT(std::equal(hist.begin(), hist.end(), expectedFrequencyStd.begin(),
+                         [](const HistogramItem &item, const double &sigma) {
+                           return item.frequencyStandardDeviation() == sigma;
+                         }));
+  }
+};
+
+class HistogramIteratorTestPerformance : public CxxTest::TestSuite {
+public:
+  static HistogramIteratorTestPerformance *createSuite() {
+    return new HistogramIteratorTestPerformance;
+  }
+  static void destroySuite(HistogramIteratorTestPerformance *suite) {
+    delete suite;
+  }
+
+  HistogramIteratorTestPerformance()
+      : m_hist(BinEdges(histSize, LinearGenerator(0, 1)),
+               Counts(histSize - 1, LinearGenerator(0, 1))) {}
+
+  void test_convert_counts_to_frequency_for_each_item() {
+    double total = 0;
+    for (size_t i = 0; i < nHists; i++)
+      for (auto &item : m_hist)
+        total += item.frequency();
+  }
+
+  void test_convert_counts_to_frequency_once_per_histogram() {
+    double total = 0;
+    for (size_t i = 0; i < nHists; i++) {
+      auto frequencies = m_hist.frequencies();
+      for (auto &frequency : frequencies)
+        total += frequency;
+    }
+  }
+
+  void test_convert_counts_to_frequency_for_each_item_sparse() {
+    double total = 0;
+    for (size_t i = 0; i < nHists; i++) {
+      for (auto &item : m_hist) {
+        if (item.counts() > histSize - 5)
+          total += item.frequency();
+      }
+    }
+  }
+
+  void test_convert_counts_to_frequency_once_per_histogram_sparse() {
+    double total = 0;
+    for (size_t i = 0; i < nHists; i++) {
+      auto counts = m_hist.frequencies();
+      auto frequencies = m_hist.frequencies();
+      for (size_t j = 0; j < nHists; ++j)
```

Comment:
`histSize`?

Also you are getting `frequencies()` twice for some reason? I suppose you wanted `counts()`?